external/mmdetection/detection_tasks/apis/detection/ote_utils.py

```diff
@@ -174,3 +175,39 @@ def on_train_end(self, logs=None):
     def on_initialization_end(self):
         self.update_progress_callback(self.loading_stage_progress_percentage +
                                       self.initialization_stage_progress_percentage)
+
+
+def get_saliency_map(feature_map: torch.Tensor) -> torch.Tensor:
+    """Generate the saliency map by normalizing the feature map to (0, 255)
+
+    Args:
+        feature_map (torch.Tensor): feature map from backbone
+
+    Returns:
+        torch.Tensor: Saliency Map
+    """
+    saliency_map = torch.sigmoid(torch.mean(feature_map, dim=1))
+    saliency_map = (
+        255
+        * (saliency_map - torch.min(saliency_map))
+        / (torch.max(saliency_map) - torch.min(saliency_map) + 1e-12)
+    )
+    saliency_map = saliency_map.to(torch.uint8)
+    return saliency_map
+
+
+def get_feature_vector(feature_maps: List[torch.Tensor]) -> torch.Tensor:
+    """Generate the feature vector by average pooling feature maps
+
+    Args:
+        feature_maps (torch.Tensor): list of feature maps from backbone
+
+    Returns:
+        torch.Tensor: feature vector(representation vector)
+    """
+    pooled_features = [
+        torch.nn.functional.adaptive_avg_pool2d(feat_map, (1, 1))
+        for feat_map in feature_maps
+    ]
+    feature_vector = torch.cat(pooled_features, dim=1)
+    return feature_vector
```

Comment:
I saw these functions in mmdet PRs, too. Do we need to keep them both?

Reply:
Fair point and removed